astroquery/mast/core.py

```diff
@@ -157,241 +155,101 @@ def __init__(self, username=None, password=None, session_token=None):
 
-            # add cookie to session
-            self._session.cookies.set_cookie(session_token)
+        if token is None and "MAST_API_TOKEN" in os.environ:
+            token = os.environ["MAST_API_TOKEN"]
 
-        elif isinstance(session_token, dict):
-            if len(session_token) > 1:
-                warnings.warn("Too many entries in token dictionary, only shibsession cookie will be used",
-                              InputWarning)
+        if token is None:
+            token = keyring.get_password("astroquery:mast.stsci.edu.token", "masttoken")
 
-            # get the shibsession cookie
-            value = None
-            for name in session_token.keys():
-                if "shibsession" in name:
-                    value = session_token[name]
-                    break
+        if token is None or reenter_token:
+            info_msg = "If you do not have an API token already, visit the following link to create one: "
+            log.info(info_msg + auth_link)
+            token = getpass("Enter MAST API Token: ")
 
-            if not value:
-                raise LoginError("Invalid session token")
+        # store token if desired
+        if store_token:
+            keyring.set_password("astroquery:mast.stsci.edu.token", "masttoken", token)
 
-            # add cookie to session
-            self._session.cookies.set(name, value)
+        self._session.headers["Accept"] = "application/json"
+        self._session.cookies["mast_token"] = token
+        info = self.session_info(silent=True)
 
+        if not info["anon"]:
+            log.info("MAST API token accepted, welcome %s" % info["attrib"].get("display_name"))
         else:
-            # raise datatype error
-            raise LoginError("Session token must be given as a dictionary or http.cookiejar.Cookie object")
+            warn_msg = ("MAST API token invalid!\n"
+                        "To make create a new API token visit to following link: " +
+                        auth_link)
+            warnings.warn(warn_msg, AuthenticationWarning)
 
-        # Print session info
-        # get user information
-        response = self._session.request("GET", self._SESSION_INFO_URL)
-
-        if response.status_code != 200:
-            warnings.warn("Status code: {}\nAuthentication failed!".format(response.status_code),
-                          AuthenticationWarning)
-            return False
-
-        exp = re.findall(r'<strong>Session Expiration \(barring inactivity\):</strong> (.*?)\n', response.text)
-        if len(exp) == 0:
-            warnings.warn("{}\nAuthentication failed!".format(response.text),
-                          AuthenticationWarning)
-            return False
-        else:
-            exp = exp[0]
+        return not info["anon"]
 
-        log.info("Authentication successful!\nSession Expiration: {}".format(exp))
-        return True
+    @deprecated(since="v0.3.10", message=("The get_token function is deprecated, "
```

Comment:
I'm not sure what to do with this deprecation. Shall we do a release after this PR is merged? Otherwise it can go on for a long time before we get a tagged 0.3.10 so this deprecation would not necessarily mean much sense to users. @keflavich?

Reply:
@ceb8 - how long do you expect the deprecated way to be working?

Reply:
I am not sure I am doing this properly. Basically because of the way authentication works now, this function doesn't make much sense and is redundant and unnecessary since the "token" is now provided by the user, so there isn't a need for them to get it back. So I want to get rid of the function.  But it seems like it shouldn't just disappear, hence the deprecation warning. How would you recommend I handle this? I can do whatever you think makes the most sense.

Reply:
Hmm, I'm not sure either. If users were using this method, then a the warning or even an exception with an explaining message is indeed seems more useful than a generic AttributeError. However as it's not doing anything it probably wouldn't need to be a deprecation.

@keflavich - preferences/ideas?

Reply:
I changed the "since" argument to v0.3.9.dev, which will hopefully indicate to the user that is deprecated as of the version they are running.

Reply:
I'm changing it to 0.3.9 as it belongs to the rest of the deprecations made in that version. And then will look into to tag a 0.3.10 release.